constraint/commitment.go

```diff
@@ -64,10 +67,67 @@ func (i *Commitment) PublicCommitted() []int {
 	return i.Committed[:i.NbPublicCommitted()]
 }
 
-func CommitmentIndexes(commitments []Commitment) []uint64 {
-	res := make([]uint64, len(commitments))
-	for i := range res {
-		res[i] = uint64(commitments[i].CommitmentIndex)
+func (c Commitments) CommitmentWireIndexes() []int {
```

Comment:
maybe evaluate if instead of computing all the index (wires, commited etc) during proving time, it would not be worth it to store these slices in the `CommitmentInfo` data struct? Particularly to avoid allocations in the prover.